lib/Runtime/Base/Utf8SourceInfo.cpp

```diff
@@ -59,6 +59,25 @@ namespace Js
 #endif
     }
 
+    LPCUTF8 Utf8SourceInfo::GetSourceFromHolder(const char16 * reason) const
+    {
+        LPCUTF8 toReturn = nullptr;
+        if (GetScriptContext()->GetThreadContext()->IsScriptActive())
+        {
+            BEGIN_LEAVE_SCRIPT(GetScriptContext())
```

Comment:
The fix I had in mind was to add these guards in DynamicSourceHolder when it makes external calls (the hits we are getting from the crawler). Are you seeing re-entrancy with JsrtSourceHolder?